lib/puppetx/puppetlabs/strings/yard/templates/default/template_helper.rb

```diff
@@ -0,0 +1,81 @@
+# A class containing helper methods to aid in the extraction of relevant data
+# from comments and YARD tags
+class TemplateHelper
+
+  # Extracts data from comments which include the supported YARD tags
+  def extract_tag_data(object)
+    examples = Hash.new
+    example_tags = object.tags.find_all { |tag| tag.tag_name == "example" }
+    example_tags.each do |example|
+      examples["#{example.name}"] = example.text
+    end
+
+    return_tag = object.tags.find { |tag| tag.tag_name == "return"}
+    return_text = return_tag.nil? ? nil : return_tag.text
+    return_types = return_tag.nil? ? nil : return_tag.types
+    return_details = (return_text.nil? && return_types.nil?) ? nil : [return_text, return_types]
+
+    since_tag = object.tags.find { |tag| tag.tag_name == "since"}
+    since_text = since_tag.nil? ? nil : since_tag.text
+
+    {:name => object.name, :desc => object.docstring, :examples => examples, :since => since_text, :return => return_details}
+  end
+
+  # Given the parameter information and YARD param tags, extracts the
+  # useful information and returns it as an array of hashes which can
+  # be printed and formatted as HTML
+  #
+  # @param parameters [Array] parameter details obtained programmatically
+  # @param tags_hash [Array] parameter details obtained from comments
+  # @param fq_name [Boolean] does this paramter have a fully qualified name?
+  #
+  # @return [Hash] The relevant information about each parameter
+  # @option opts [String] :name The name of the parameter
+  # @option opts [String] :fq_name The fully qualified parameter name
+  # @option opts [String] :desc The description provided in the comment
+  # @options opts [Array] :types The parameter type(s) specified in the comment
+  # @options opts [Boolean] :exists? True only if the parameter exists in the documented logic and not just in a comment
```

Comment:
yeah, for a parameter, but you are using options to describe the returned hash - do not think that works. Each @option has a reference to the @param it describes a detail of. There is no way I can see to make it refer to the returned value.